pkg/network/tracer/ebpf_conntracker.go

```diff
@@ -48,20 +48,22 @@ var tuplePool = sync.Pool{
 	},
 }
 
-type ebpfConntrackerStats struct {
-	gets                 *atomic.Int64
-	getTotalTime         *atomic.Int64
-	unregisters          *atomic.Int64
-	unregistersTotalTime *atomic.Int64
-}
-
-func newEbpfConntrackerStats() ebpfConntrackerStats {
-	return ebpfConntrackerStats{
-		gets:                 atomic.NewInt64(0),
-		getTotalTime:         atomic.NewInt64(0),
-		unregisters:          atomic.NewInt64(0),
-		unregistersTotalTime: atomic.NewInt64(0),
-	}
+const ebpfConntrackerModuleName = "network_tracer__ebpf_conntracker"
+
+var defaultBuckets = []float64{10, 25, 50, 75, 100, 250, 500, 1000, 10000}
+
+var conntrackerTelemetry = struct {
+	getsDuration        telemetry.Histogram
+	unregistersDuration telemetry.Histogram
+	getsTotal           telemetry.Gauge
+	registersTotal      telemetry.Gauge
+	unregistersTotal    telemetry.Gauge
+}{
+	telemetry.NewHistogram(ebpfConntrackerModuleName, "gets_duration_nanoseconds", []string{}, "Histogram measuring the time spent retrieving connection tuples from the EBPF map", defaultBuckets),
+	telemetry.NewHistogram(ebpfConntrackerModuleName, "unregisters_duration_nanoseconds", []string{}, "Histogram measuring the time spent deleting connection tuples from the EBPF map", defaultBuckets),
+	telemetry.NewGauge(ebpfConntrackerModuleName, "gets_total", []string{}, "Gauge measuring the total number of attempts to get connection tuples from the EBPF map"),
+	telemetry.NewGauge(ebpfConntrackerModuleName, "registers_total", []string{}, "Gauge measuring the total number of attempts to update/create connection tuples in the EBPF map"),
+	telemetry.NewGauge(ebpfConntrackerModuleName, "unregisters_total", []string{}, "Gauge measuring the total number of attempts to delete connection tuples from the EBPF map"),
```

Comment:
These seem like counters, WDYT?

Reply:
agreed except for `registers_total` which may still have to be a gauge since it is setting the value to what is pulled from the ebpf map:

```
telemetry := &netebpf.ConntrackTelemetry{}
if err := e.telemetryMap.Lookup(unsafe.Pointer(&zero), unsafe.Pointer(telemetry)); err != nil {
   log.Tracef("error retrieving the telemetry struct: %s", err)
} else {
   conntrackerTelemetry.registersTotal.Set(float64(telemetry.Registers))
}
```

the counter interface only has `Inc` and `Add`, neither of which really work here, unless we kept a `lastRegisters` value and `Add`ed the delta. I think a gauge seems simpler, even if the value is monotonically increasing. what is your take?

Reply:
That makes sense. I think the least confusing thing from a metrics point of view would be to make a it counter and compute the delta like you said. Otherwise when you are graphing them, `registers` is going to behave quite different from `unregisters`.